### PR TITLE
new readme

### LINKED ISSUE #2: re-purpose for sinatra track
add definition of sessions, relationship btwn cookies and session. 

http://stackoverflow.com/questions/3804209/what-are-sessions-how-do-they-work

deadline: 9/15/15

### PR DESCRIPTION
**Pull request by SophieDeBenedetto**
needs :eyes: 

@dfenjves @ipc103 @deniznida @AnnJohn 

closes #2

### CODE REVIEW

**Comment by AnnJohn** [comment]
Assigned to @deniznida for review. Deadline today, 9/11/15

File: README.md [hunk 1]
```diff
@@ -1,30 +1,55 @@
...
+
+> HTTP is a stateless protocol, treating each request as an independent transaction that is unable to use information from any previous requests. This means there is no way within the hypertext transfer protocol to remember a user’s identity from page to page; instead, web applications requiring user login must use a session, which is a semi-permanent connection between two computers (such as a client computer running a web browser and a server running Rails).
+
+>–– Michael Hartl, *The Ruby on Rails Tutorial*
+
+A **session** is simply an object, like a hash, that stores data describing a client's interaction with a website at a given point in time. The session hash lives on the server. Your application can access it in any of your controllers at any point in time. 
+
+How does a session know who is interacting with a website? Through **cookies**. Cookies are are snippets of text placed on the user’s browser. Cookies persist from one page to the next when you are browsing the web. So, they can store information (such as a user id) that can then be used by the application to retrieve the logged-in user from the database, among other things.
```
Comment (deniznida) [inline README.md]: 1. doesn't a server know who is interacting with a website, by using cookies and session data
2. should we add session id to this sentence --So, they can store information (such as a user id and session id)

File: README.md [hunk 2]
```diff
@@ -1,30 +1,55 @@
...
+How does a session know who is interacting with a website? Through **cookies**. Cookies are are snippets of text placed on the user’s browser. Cookies persist from one page to the next when you are browsing the web. So, they can store information (such as a user id) that can then be used by the application to retrieve the logged-in user from the database, among other things. 
+
+## How do Cookies Work?
+
+Here's the general flow: 
+
+1. You visit a webpage, www.facebook.com 
+2. The web application that is Facebook receives that request and creates a cookie with some information about you as a user. 
+3. Facebook sends that cookie *back to the browser* where it is stored temporarily. The cookie will last until you log out of Facebook, close your browser or clear your cache. 
+4. Every subsequent request you, the client, sends to www.facebook.com sends the cookies *back to the server* where your application can access it, get information out of and alter it.
```
Comment (deniznida) [inline README.md]: would it make sense to change this sentence to:

Every subsequent request a client sends to www.facebook.com, sends the cookies _back to the server_ where the facebook application can access it, get information out of and alter it.

File: README.md [hunk 3]
```diff
@@ -1,30 +1,55 @@
...
+## Coming Up...
 
-Persistent cookies allow the website to remember your user information and preferences for future visits. This allows for faster page loading, automatic login and user authentication, and access to other potential web application features. Persistent cookies are usually created on the first visit to the page after you have created an account, and typically don't expire unless the web application has a protocol in place for cookie expiration, or the user clears their browser's cookie cache.
+So far, we've defined a **session** as a hash that lives on the server side of our application and stores information about a particular user who is interacting with our website at a given point in time. The session object will get information from and interact with cookies. Later on, we'll learn how to use the session object to build apps that allow users to log in and log out.
```
Comment (deniznida) [inline README.md:62]: Does this make sense?

Cookies are client side counterparts to sessions. They store information on client computers and and are visible only to the server that created them. Client browsers send cookies to remote servers along with each request, then the server can read the cookie, associate it with an existing session if any and decides how to respond accordingly. (e.g. if there is no cookie received: show login page, cookie received but doesn't match a session: show login page remembering user name, cookie received and matches a session : respond with data). 
Later on, we'll learn how to use the cookie and session objects to build apps that allow users to log in and log out.

**Comment by deniznida** [comment]
@SophieDeBenedetto added some inline comments

**Comment by SophieDeBenedetto** [comment]
addressed @deniznida comments. Thanks, super helpful!

merging now.

**Pull request merged**